crates/shared/src/price_estimation.rs

```diff
@@ -133,6 +134,24 @@ pub async fn ensure_token_supported(
     }
 }
 
+#[mockall::automock]
+#[async_trait::async_trait]
+pub trait NativePriceEstimating: Send + Sync {
```

Comment:
Nit: I feel this fits better in the `native` module / file too next to the implementation but not sure.